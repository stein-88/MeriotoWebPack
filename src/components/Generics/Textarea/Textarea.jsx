import React from 'react'

const Textarea = ({
 placeholder, value, onChange, id,
}) => {
    if (!id || !onChange) return null
    return (
        <textarea
            className="w-100"
            placeholder={placeholder}
            value={value || ''}
            onChange={(ev) => onChange(ev)}
            id={id}
        />
    )
}

export default Textarea
