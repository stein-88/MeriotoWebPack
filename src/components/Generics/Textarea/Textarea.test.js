import React from 'react'
import renderer from 'react-test-renderer'
import Textarea from './Textarea'

describe('Should not render without props', () => {
    test('Should not render without props', () => {
        const compTextarea = renderer.create(<Textarea />)
        const jsonTextarea = compTextarea.toJSON()
        expect(jsonTextarea).toBeNull()
    })
    test('Should not render with onChange only', () => {
        const compTextarea = renderer.create(<Textarea onChange={() => null} />)
        const jsonTextarea = compTextarea.toJSON()
        expect(jsonTextarea).toBeNull()
    })
    test('Should not render with id only', () => {
        const compTextarea = renderer.create(<Textarea id="dsaf32rw" />)
        const jsonTextarea = compTextarea.toJSON()
        expect(jsonTextarea).toBeNull()
    })
    test('Should render with full props', () => {
        const mockFunc = jest.fn()
        const compTextarea = renderer.create(
            <Textarea
                onChange={() => null}
                id="dsaf32rw"
                placeholder="sdfsda"
            />
        )
        const jsonTextarea = compTextarea.toJSON()
        jsonTextarea.props.onChange()
        expect(mockFunc).not.toHaveBeenCalled()
        expect(jsonTextarea).toMatchSnapshot()
    })
})