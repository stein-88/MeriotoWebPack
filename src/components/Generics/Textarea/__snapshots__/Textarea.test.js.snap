// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should not render without props Should render with full props 1`] = `
<textarea
  className="w-100"
  id="dsaf32rw"
  onChange={[Function]}
  placeholder="sdfsda"
  value=""
/>
`;
